src/cmd/serve/async_index.rs

```diff
@@ -33,8 +39,14 @@ impl Indexes {
         let mut srch = BTreeMap::new();
         let insert_search_entry = |index: &mut BTreeMap<_, _>, entry: RecipeEntry| {
             let content = entry.read().expect("can't read recipe");
-            let meta = content.metadata(&parser);
-            index.insert(entry.path().to_owned(), meta);
+            index.insert(
+                entry.path().to_owned(),
+                RecipeData {
+                    metadata: content.metadata(&parser),
+                    ingredients: content.ingredients(&parser),
+                    cookware: content.cookware(&parser),
```

Comment:
You are parsing the recipe 3 times here: 1 for metadata and 2 full parses. You can just parse it once and extract all the information.